### PR TITLE
Improving Categorical Distribution Docs' (#16291)

### PR DESCRIPTION
**Pull request by JamesTrick**
**Closes:** Confusing documentation with distributions.Categorical about logits. Closes #16291

**Solution**: Changes documentation on the Categorical distribution from `log probabilities` to `event log-odds`. This makes should reduce confusion as raised by this issue, and is consistent with other distributions such as `torch.Binomial`. 

More than happy to make any other changes if they fit :).

### CODE REVIEW

**Review by soumith — APPROVED** [review]
thanks!
